### DISCUSSION TITLE
How to preload translations on app start-up rather than first request?

### DISCUSSION THREAD

**Discussion by andygnewman** [i18next/next-i18next#1622] [Q&A]
I'm working with an app with 25 locales and lots of namespaces.
The way the app has been constructed, mainly following the project setup pattern described in the README, it loads all translations on the first page request.

With so many translations I would like to have them preloaded on server start-up so they are not tied to a specific request, but I'm struggling to understand how that can be done?

Is there an example available for this? Apologies if I've missed something obvious in the documentation.

**Comment by isaachinman** [comment]
The translations are loaded when `serverSideTranslations` is called. There is no "singleton" translation instance model – each request has its own i18next client initialised.

Are you experiencing some sort of latency?

**Reply by andygnewman** [reply]
Thanks for the quick response.

We're seeing some latency of upto 2000ms caused by readFile operations on the initial serverSideTranslations request.

**Reply by andygnewman** [reply]
So I could do something like;
```
await serverSideTranslations(defaultLocale, []);
```
in the server.js, ahead of `app.listen(port)`, which triggers the fileReads.

**Reply by isaachinman** [reply]
I assume you are using `getServerSideProps`, not `getStaticProps`, then. 2000ms latency for fs reads raises an eyebrow, and sounds like an issue on its own.

Back to your original question, it is not possible to share a singleton translation instance in the NextJs render process, as individual requests are handled by a worker pool and do not necessarily share memory. This is true of both the SSR and SSG processes.

Your code sample seems to indicate you are running an express server, is that the case? Custom servers are very much deprecated at this point, to my understanding.

**Reply by andygnewman** [reply]
Yes we are using getServerSideProps.

And it is an express server that we're running.

It might well be that there is another factor impacting the readFile latency, but our first priority for analysis of the latency was to understand what was triggering the readFile operations.
